src/Kiota.Builder/Plugins/PluginsGenerationService.cs

```diff
@@ -112,6 +113,40 @@ public async Task GenerateManifestAsync(CancellationToken cancellationToken = de
         }
     }
 
+    private static void EnsureSecuritySchemeExtensions(OpenApiDocument document)
+    {
+        var securitySchemes = document?.Components?.SecuritySchemes;
+        if (securitySchemes != null)
+        {
+            foreach (var securitySchemeItem in securitySchemes)
+            {
+                string? authenticationReferenceId = null;
+                var securityScheme = securitySchemeItem.Value;
+                var securitySchemeName = securitySchemeItem.Key;
+                if (securityScheme != null)
+                {
+                    if (securityScheme.Extensions is not null && securityScheme.Extensions.TryGetValue(OpenApiAiAuthReferenceIdExtension.Name, out var authReferenceIdExtension) && authReferenceIdExtension is OpenApiAiAuthReferenceIdExtension authReferenceId)
+                        authenticationReferenceId = authReferenceId.AuthenticationReferenceId;
+                    if (authenticationReferenceId == null)
+                    {
+                        var auth = TryGetAuthFromSecurityScheme(securitySchemeName, securityScheme);
+                        if (auth is not null)
```

Comment:
use type assertion in condition

Reply:
changes done in commit 25390f69335a38955531b39922d537f51a0b0477